buildkite/builds.go

```diff
@@ -113,6 +128,9 @@ type BuildsListOptions struct {
 	// Include all retried jobs in each build’s jobs list
 	IncludeRetriedJobs bool `url:"include_retried_jobs,omitempty"`
 
+	// Filters results by metadata.
+	MetaData MetaDataFilters
```

Comment:
@andrewhamon I believe what we would require here (after testing) would be:

```
MetaData MetaDataFilters `url:"meta_data,omitempty"`
```

By default - the [encoded](https://github.com/google/go-querystring/blob/master/query/encode.go#L57-L74) URL key when calling `encodeValues` defaults to the struct's name, unless specified with the `url` tag value as above. In our case - `MetaData` is passed as the URL parameter (which of course needs to be `meta_data` to call the builds API successfully with that parameter!